Source/WebCore/testing/MockGamepadProvider.cpp

```diff
@@ -87,8 +95,15 @@ bool MockGamepadProvider::connectMockGamepad(unsigned index)
 
     m_connectedGamepadVector[index] = m_mockGamepadVector[index].get();
 
-    for (auto& client : m_clients)
+    for (auto& client : m_clients) {
         client->platformGamepadConnected(*m_connectedGamepadVector[index], EventMakesGamepadsVisible::Yes);
+        auto gamepadsForClient = m_invisibleGamepadsForClient.find(client);
+        if (gamepadsForClient != m_invisibleGamepadsForClient.end()) {
+            for (auto& invisibleGamepad : gamepadsForClient->value)
+                client->platformGamepadConnected(*invisibleGamepad, EventMakesGamepadsVisible::Yes);
+            m_invisibleGamepadsForClient.remove(client);
```

Comment:
Seems like this should be `remove(gamepadsForClient)` to avoid an extra hash map look up.